logstash-core/src/main/java/org/logstash/execution/QueueReadClientBase.java

```diff
@@ -131,13 +130,6 @@ public IRubyObject rubyReadBatch(final ThreadContext context) throws Interrupted
     public void closeBatch(QueueBatch batch) throws IOException {
         batch.close();
         inflightBatches.remove(Thread.currentThread().getId());
-        Long startTime = inflightClocks.remove(Thread.currentThread().getId());
```

Comment:
review note: when closing a batch, we no longer do manual time math, and are no longer responsible for removing a start time from the map because execution is wrapped in `executeWithTimers(Runnable)`.